lib/ecto/adapters/postgres/connection.ex

```diff
@@ -251,24 +251,55 @@ if Code.ensure_loaded?(Postgrex) do
     end
 
     @impl true
-    def explain_query(query, opts \\ []) do
-      verbose = Keyword.get(opts, :verbose, false) |> expr(nil, nil)
-      costs   = Keyword.get(opts, :costs,   true)  |> expr(nil, nil)
-      buffers = Keyword.get(opts, :buffers, false) |> expr(nil, nil)
-
-      analyze =
-        case buffers do
-          "TRUE" -> "TRUE"
-          "FALSE" -> Keyword.get(opts, :analyze, false) |> expr(nil, nil)
+    def explain_query(query, server_version, opts \\ []) do
+      server_version = normalize_version(server_version)
+
+      if Version.compare(server_version, "9.0.0") == :lt do
+        raise RuntimeError, "version not supported"
```

Comment:
Do we need to support older versions?

Reply:
See my comment above about supporting multiple versions. We can probably just move the complexity to the caller. Then we have a single integration test and use unit tests to test the different options.

Reply:
Fixed. It supports all versions now, and it's up to the developer to pass the right options.